src/satosa/exception.py

```diff
@@ -30,6 +30,39 @@ class SATOSACriticalError(SATOSAError):
     """
     pass
 
+class SATOSAProcessingHaltError(SATOSAError):
+    """
+    SATOSA error that should stop processing
+    """
+    def __init__(self, state, message, *args, **kwargs):
+        """
+        :type state: satosa.state.State
+        :param message: str
+        :param args: any
+        :param kwargs: any
+
+        :param state: Satosa state
+        :param message: A test message
+        :param args: whatever
+        :param kwargs: whatever
+        """
+        super().__init__(message, *args, **kwargs)
+        self._message = "One of the plugins or microservices raised an error. Redirect to an error page"
```

Comment:
The place where the exception is caught is very generic. Can the core (not through a front/backend module or microservice) throw this error? 
If yes, then I think it is fine, but the message should also be more generic like `Request processing stopped due an error. Redirecting to an error page..`, as the error could originate from the core. 
If not, then the message is fine, but I would move the exception handling around the places where the error would occur (ie around the calls to `process()` for the microservices). At first I didn't understand that by _plugins_ you meant the front/backend modules, as in my head microservices are kindof like plugins. Finding and wrapping calls to front/backend modules will make things ugly, as those are scattered around the codebase.
If microservices and front/backend modules had the same interface, then this could be refactored to wrap a single call to their entry point. So, this is not something strict, just a suggestion. 
Placing the exception handling under `run()` is very convenient, but it should not be overused - in other words the following for example, is also very convenient but I would not recommend it:

```python
if __name__ == "__main__":
  try:
    main()
  except SomeVerySpecificError as e:
    log('some very specific message')
    handle_specific_error(e)
  except SomeError as e:
    log('some message')
    handle_some_error(e)
```

Errors should be caught and handled by the callers that triggered their originator. If that level is not appropriate then the exception is propagated up the stack until a suitable level - with the right contextual information for the error - can handle it (the very specific error above is most probably better handled in a deeper level). So, I would first try to find that level, than using the top level for every error. 

Now, for our case, this error is generic enough to fit this place, and I will not object having it there ;) 

But then, I am thinking, why shouldn't we redirect to an error page, for any error that is not handled deeper? ie, why should `SATOSANoBoundEndpointError`s not also be redirected to a page where the error message is shown in a nice/user-friendly way? (This would require we build a template/html-page to display errors)